controllers/metric_controller.go

```diff
@@ -0,0 +1,256 @@
+							ServerName: &serverName,
+						},
+					},
+					RelabelConfigs: []monv1.RelabelConfig{
+						{
+							Action:      "replace",
+							Regex:       "(.*)",
+							Replacement: &replacement0,
+							TargetLabel: "instance",
+							SourceLabels: []monv1.LabelName{
+								"__meta_kubernetes_endpoint_address_target_name",
+							},
+						},
+						{ // Include only Windows nodes for this serviceMonitor
+							Action: "keep",
+							Regex:  "windows",
+							SourceLabels: []monv1.LabelName{
+								"__meta_kubernetes_node_label_kubernetes_io_os",
+							},
+						},
+						{ // Change the port from the kubelet port 10250 to 9182
+							Action:      "replace",
+							Regex:       "(.+)(?::\\d+)",
+							Replacement: &replacement1,
+							TargetLabel: "__address__",
+							SourceLabels: []monv1.LabelName{
+								"__address__",
+							},
+						},
+						{ // Update the job label from kubelet to windows-exporter
+							Action:      "replace",
+							Replacement: &replacement2,
+							TargetLabel: "job",
+						},
+					},
+				},
+			},
+			NamespaceSelector: monv1.NamespaceSelector{
+				MatchNames: []string{"kube-system"},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"k8s-app": "kubelet",
+				},
+			},
+		},
+	}
+
+	if err == nil {
+		// check if existing serviceMonitor's contents are as expected, delete it if not
+		if existingSM.Name == expectedSM.Name && existingSM.Namespace == expectedSM.Namespace &&
+			reflect.DeepEqual(existingSM.Spec, expectedSM.Spec) {
+			return nil
+		}
+		err = r.ServiceMonitors(r.watchNamespace).Delete(context.TODO(), metrics.WindowsMetricsResource,
+			metav1.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to delete service monitor %s/%s: %w", r.watchNamespace, metrics.WindowsMetricsResource,
+				err)
+		}
+		r.log.Info("Deleted malformed resource", "serviceMonitor", metrics.WindowsMetricsResource,
+			"namespace", r.watchNamespace)
+	}
+
+	_, err = r.ServiceMonitors(r.watchNamespace).Create(context.TODO(), expectedSM, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("error creating service monitor: %w", err)
+	}
+	return nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *metricReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	metricsPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isMonitoringEnabled(e.Object, r.watchNamespace)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isMonitoringEnabled(e.ObjectNew, r.watchNamespace)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return isMonitoringEnabled(e.Object, r.watchNamespace)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1.Namespace{}, builder.WithPredicates(metricsPredicate)).
+		Watches(&monv1.ServiceMonitor{}, &handler.EnqueueRequestForObject{}).
```

Comment:
This is going to pass the servicemonitor object to the Reconcile function. This is also going to bypass any predicates. Lets be consistent on the object being reconciled. You could use `EnqueueRequestsFromMapFunc` to map the servicemonitor to the namespace it is in.